discovery.md

```diff
@@ -199,6 +202,18 @@ with the wire format/API defined by this specification.
 The following sections define the attributes that appear in a Service
 entity.
 
+##### id
+- Type: `String`
+- Description: A unique identifier for this Service. This value MUST be
+  globally unique. While other metadata within this Service MAY change,
+  this value MUST NOT. Clients can use this value to ensure that the Service
+  entity returned from a query refers to a previously retrieved Service.
```

Comment:
I wonder about the lifecycle of this id. Does the UUID have to be the same, if discovery of the same service is possible via different discovery endpoints? Who is responsible for assigning this id?

Reply:
I'm leaning towards saying a little as possible on this to allow for the most flexibility. Meaning, since it must be globally unique, then if the implementation chooses to use the same ID across different discovery endpoints, then they are basically telling all clients that it is the same service. If they then also choose to change the metadata about that service based on the discovery endpoint, then the spec is silent as to what that means - beyond they are the same service.  So, whether one endpoint's metadata should be used in place of another one's metadata, or whether the metadata is supposed to be different because each endpoint is assuming a different type of user is doing the query, is out of scope for the spec to decide.

wdyt?  Is this something we can resolve in a follow-on PR or do you think it needs to be nailed down now?

Reply:
@duglin I wonder, if this metadata belongs to the discovery endpoint or the service. Does the ID identify the service or the service entry in a specific discovery endpoint? We might also defer this decision until later, when we understand better, how things will be used and interact. We might not be able to clarify this directly with a follow-on PR, but we could accept the PR and create an issue to remind us later about this question.